### PR TITLE
Add tests for template and cleanup unused code

### PR DESCRIPTION
**Pull request by geek**
* add template tests
* add tests for validating hapi options
* removed code that was unused and appears to not be needed by hapi

### CODE REVIEW

**Review by cjihrig — APPROVED** [review]

**Comment by devinivy** [comment]
> add tests for validating hapi options

I think this is going to be very useful, great idea.

> removed code that was unused and appears to not be needed by hapi

Validate will be used for validation within other hapi modules too, so I think hapi makes for a great smoke test but perhaps not a perfect test for whether a line needs to be kept or not.

File: test/template.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    it('isTemplate returns false when not a template', () => {
+
+        expect(Template.isTemplate(null)).to.be.false();
+    });
+
+    it('_parse will parse {} templates', () => {
+
+        const template = new Template('foo {raw} \\{esc\\}{');
+        template._parse();
```
Comment (devinivy) [inline test/template.js:33]: I believe `_parse()` is called by the constructor and can probably be omitted here if you like.

File: test/template.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        const state = {
+            path: 'test/',
+            ancestors: ['test'],
+            mainstay: {}
+        };
+
+        const rendered = template.render('test', state, null, null, { errors: { escapeHtml: true } });
+
+        expect(rendered).to.equal('foo  {esc}');
```
Comment (devinivy) [inline test/template.js:54]: The original template had a trailing `{` which I would expect to appear here.

Comment (geek) [reply test/template.js:54]: The current behavior is to strip any unescaped trailing `{`

**Review by devinivy — APPROVED** [review]

**Pull request merged**